### PR TITLE
Extend docs about `train!`

### PR DESCRIPTION
**Pull request by heliosdrm**
Related to #921: explain why it is not needed to pass the model as argument.

### CODE REVIEW

File: docs/src/training/training.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ With these we can call `Flux.train!`:
 ```julia
 Flux.train!(objective, params, data, opt)
 ```
+At first glance it may seem strange that the model that we want to train is not part of the input arguments of `Flux.train!`. However the target of the optimizer is not the model itself, but the objective function that represents the departure between modelled and observed data. In other words, the model is implicitly defined in the objective function, and there is no need to give it explicitly. Passing the objective function instead of the model and a cost function separately (see below) provides more flexibility, and the possibility of optimizing the calculations.
```
Comment (MikeInnes) [inline docs/src/training/training.md]: I think this should be the first paragraph of the "loss functions" section just below, but otherwise LGTM.

Comment (heliosdrm) [reply docs/src/training/training.md]: Really the first paragraph, or maybe rather the last one?

Comment (MikeInnes) [reply docs/src/training/training.md]: Sure, that works well too.

Comment (heliosdrm) [reply docs/src/training/training.md]: Moved to the suggested place.

File: docs/src/training/training.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Training
 
-To actually train a model we need three things:
+To actually train a model we need three things, in addition to the tracked parameters that will be fitted:
```
Comment (MikeInnes) [inline docs/src/training/training.md]: This should probably just be changed to "four things" and list parameters in the right place.

Comment (heliosdrm) [reply docs/src/training/training.md]: Made as suggested, and also included a sub-section about the params argument. There is not a sub-section about the optimiser, but I guess this is already covered by the link to the page dedicated to it.

File: docs/src/training/training.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
 # Training
 
-To actually train a model we need three things:
+To actually train a model we need four things:
 
 * A *objective function*, that evaluates how well a model is doing given some input data.
+* The parameters of the model.
```
Comment (MikeInnes) [inline docs/src/training/training.md]:
```suggestion
* The trainable parameters of the model.
```

**Comment by MikeInnes** [comment]
Thanks @heliosdrm!

**Pull request merged**